cmake/external/ObsPluginHelpers.cmake

```diff
@@ -1,163 +1,348 @@
-# Functions for generating external plugins
+if(POLICY CMP0087)
+  cmake_policy(SET CMP0087 NEW)
+endif()
 
-set(EXTERNAL_PLUGIN_OUTPUT_DIR "${CMAKE_BINARY_DIR}/rundir")
+set(OBS_STANDALONE_PLUGIN_DIR "${CMAKE_SOURCE_DIR}/release")
+set(INCLUDED_LIBOBS_CMAKE_MODULES ON)
 
-# Fix XCode includes to ignore warnings on system includes
-function(target_include_directories_system _target)
+include(GNUInstallDirs)
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "Darwin")
+  set(OS_MACOS ON)
+  set(OS_POSIX ON)
+elseif("${CMAKE_SYSTEM_NAME}" MATCHES "Linux|FreeBSD|OpenBSD")
+  set(OS_POSIX ON)
+  string(TOUPPER "${CMAKE_SYSTEM_NAME}" _SYSTEM_NAME_U)
+  set(OS_${_SYSTEM_NAME_U} ON)
+elseif("${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+  set(OS_WINDOWS ON)
+  set(OS_POSIX OFF)
+endif()
+
+# Old-Style plugin detected, find "modern" libobs variant instead and set global
+# include directories to fix "bad" plugin behaviour
+if(DEFINED LIBOBS_INCLUDE_DIR AND NOT TARGET OBS::libobs)
+  message(
+    DEPRECATION
+      "You are using an outdated method of adding 'libobs' to your project. Refer to the updated wiki on how to build and export 'libobs' and use it in your plugin projects."
+  )
+  find_package(libobs REQUIRED)
+  if(TARGET OBS::libobs)
+    set_target_properties(OBS::libobs PROPERTIES IMPORTED_GLOBAL TRUE)
+    message(STATUS "OBS: Using modern libobs target")
+
+    add_library(libobs ALIAS OBS::libobs)
+    if(OS_WINDOWS)
+      add_library(w32-pthreads ALIAS OBS::w32-pthreads)
+    endif()
+  endif()
+endif()
+
+if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT AND OS_WINDOWS)
+  set(CMAKE_INSTALL_PREFIX
+      "${OBS_STANDALONE_PLUGIN_DIR}"
+      CACHE STRING "Directory to install OBS plugin after building" FORCE)
+endif()
+
+if(NOT CMAKE_BUILD_TYPE)
+  set(CMAKE_BUILD_TYPE
+      "RelWithDebInfo"
+      CACHE STRING
+            "OBS build type [Release, RelWithDebInfo, Debug, MinSizeRel]" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS Release RelWithDebInfo
+                                               Debug MinSizeRel)
+endif()
+
+file(RELATIVE_PATH RELATIVE_INSTALL_PATH "${CMAKE_SOURCE_DIR}"
+     "${CMAKE_INSTALL_PREFIX}")
+file(RELATIVE_PATH RELATIVE_BUILD_PATH "${CMAKE_SOURCE_DIR}"
+     "${CMAKE_BINARY_DIR}")
+
+# Set-up OS-specific environment and helper functions
```

Comment:
"Set up" is the verb form, as mentioned elsewhere.

Reply:
Fixed.